packages/generator/src/entity/namespace.ts

```diff
@@ -31,14 +31,17 @@ function properties(entity: VdmEntity): VariableStatementStructure[] {
 
 function property(prop: VdmProperty, entity: VdmEntity): VariableStatementStructure {
   const type = `'${prop.edmType.startsWith('Edm') ? prop.edmType : prop.edmType.split('.').pop()}'`;
+  const initilizer = prop.isComplex
+    ? `new ${prop.fieldType}('${prop.originalName}', ${entity.className})`
+    : `new ${prop.fieldType}('${prop.originalName}', ${entity.className}, ${type})`;
   return {
     kind: StructureKind.VariableStatement,
     declarationKind: VariableDeclarationKind.Const,
     declarations: [
       {
         name: prop.staticPropertyName,
         type: `${prop.fieldType}<${entity.className}>`,
-        initializer: `new ${prop.fieldType}('${prop.originalName}', ${entity.className}, ${type})`
+        initializer: initilizer
```

Comment:
If the suggestion above is accepted, this can be re-written as:
```suggestion
        initializer
```